test/common/index.js

```diff
@@ -759,3 +759,19 @@ exports.getTTYfd = function getTTYfd() {
   }
   return tty_fd;
 };
+
+// Hijack stdout and stderr
+function hijackStdWritable(name, listener) {
+  const stream = process[name];
+  const _write = stream.write.bind(stream);
+
+  stream.writeTimes = 0;
+  stream.write = function(data) {
+    listener(data);
+    _write(data);
+    stream.writeTimes++;
```

Comment:
In fact, `listener` is something like `.emit('write')` but not `.emit('written')`, so it's sync.

And I've added the `callback` in `_write` now.